tests/exception/t13115.nim

```diff
@@ -1,12 +1,31 @@
-discard """
-  exitcode: 1
-  targets: "c"
-  matrix: "-d:debug; -d:release"
-  outputsub: ''' and works fine! [Exception]'''
-"""
+const msg = "This char is `" & '\0' & "` and works fine!"
 
-# bug #13115
-# xxx bug: doesn't yet work for cpp
-
-var msg = "This char is `" & '\0' & "` and works fine!"
-raise newException(Exception, msg)
+when defined nim_t13115:
+  # bug #13115
+  template fn =
+    raise newException(Exception, msg)
+  when defined nim_t13115_static:
+    static: fn()
+  fn()
+else:
+  import std/[osproc,strformat,os,strutils]
+  proc main =
+    const nim = getCurrentCompilerExe()
+    const file = currentSourcePath
+    for b in "c js cpp".split:
+      when defined(openbsd):
+        if b == "js":
+          # xxx bug: pending #13115
+          # remove special case once nodejs updated >= 12.16.2
+          # refs https://github.com/nim-lang/Nim/pull/16167#issuecomment-738270751
+          continue
+      for opt in ["-d:nim_t13115_static", ""]:
+        let cmd = fmt"{nim} r -b:{b} -d:nim_t13115 {opt} --hints:off {file}"
```

Comment:
Please provide an option to switch between `-d:debug` and `-d:release` mode. Because the behaviour differs.

Reply:
see https://github.com/nim-lang/Nim/blob/d5ea1cd3ffcd65f97fcca22a4e47611e4f905a23/lib/system/excpt.nim#L151

Reply:
```ini
@if release or danger:
  stacktrace:off
  excessiveStackTrace:off
  linetrace:off
  debugger:off
  line_dir:off
  opt:speed
  define:release
@end
```

Reply:
done, and also took care to avoid combinatorial explosion